### PR TITLE
go/control: Add per-runtime status report

### LINKED ISSUE #3234: Add runtime status to node status response
The node status returned by the `GetStatus` control endpoint should be updated to also include a map of runtimes supported by the running node and then for each runtime at least the following information:
* Genesis round, timestamp and state root.
* Latest round, timestamp and state root.
* Flag whether the node is currently in the executor committee.
* Flag whether the node is currently in the storage committee.
* If the node is a storage node, it should also include the information about the latest synced round.

### PR DESCRIPTION
**Pull request by kostko**
Fixes #3234

### CODE REVIEW

File: .changelog/3234.feature.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+go/control: Add per-runtime status report
+
+An additional `runtimes` field has been added to the output of the
```
Comment (ptrus) [inline .changelog/3234.feature.md:3]: Might be nice to also update the example command docs, with a runtime node example https://github.com/oasisprotocol/oasis-core/blob/master/docs/oasis-node/cli.md

**Review by ptrus — APPROVED** [review]

**Review by abukosek — APPROVED** [review]

**Comment by ptrus** [comment]
Should we also add a CLI E2E test for this command (similarly as we have stake/registry/identity cli e2e tests). (could be that we already do this, but i didn't see it in a quick search, only some indirect testing of it in `e2e/node_shutdown`)

File: .changelog/3234.feature.md (hunk 1 above)
Comment (kostko) [reply .changelog/3234.feature.md:3]: Done.

**Comment by kostko** [comment]
So we do test the `GetStatus` API call (in the [queries txsource workload](https://github.com/oasisprotocol/oasis-core/blob/99f3bcede29339f2a0cda9acd94dbddd71e3e590/go/oasis-node/cmd/debug/txsource/workload/queries.go#L658)), but not the CLI subcommand itself.

**Review by Yawning — APPROVED** [review]

**Pull request merged**